src/main/java/dk/kb/image/TemplateMostUsedColors.java

```diff
@@ -0,0 +1,136 @@
+        // Create combined map of buckets and bucket count
+        Map<C, Integer> bucketsWithCount = combineBucketsAndBucketCount(buckets, bucketCount);
+        // Sort the map
+        List<Map.Entry<C, Integer>> sortedList = sortList(bucketsWithCount);
+        // Return results
+        return returnResult(sortedList, x);
+    }
+
+    /**
+     * Defines which color palette to use and in which colorspace it should be represented.
+     * @return the palette as a list of given type.
+     */
+    abstract List<C> defineBuckets();
+
+    /**
+     * Loop through pixels of input image, get color in colorspace defined by subclass for pixel and +1 to bucket closest to pixel color.
+     * @param buckets list of bucket colors, type is defined by subclass.
+     * @return the integer array bucketCounter, which contains the count for each bucket for the input image,
+     */
+    int[] getBucketCount(BufferedImage img, List<C> buckets){
+        // Create bucket counter array
+        int[] bucketCounter = new int[buckets.size()];
+        // Loop over all pixels in image and get RGB color
+        int height = img.getHeight();
+        int width = img.getWidth();
+        for(int y = 0; y < height; y++) {
+            for (int x = 0; x < width; x++) {
+                // Get RGB for each pixel
+                pixelCount ++;
+                int pixelRGB = img.getRGB(x, y);
+                updateBucketCounter(pixelRGB, buckets, bucketCounter);
+            }
+        }
+        return bucketCounter;
+    }
+
+    /**
+     * Method to update bucketCounter in getBucketCount().
+     * @param pixel The current pixels color value in RGB colorspace.
+     * @param buckets List of color buckets of used type in the implementation.
+     * @param bucketCounter Integer array to store count of buckets.
+     */
+    void updateBucketCounter(int pixel, List<C> buckets, int[] bucketCounter) {
+        // Convert RGB pixel value to OK Lab float
+        // Values for checking max
+        int bestColor = 0;
+        double minDistance = 2147483647;
+        for (int i = 0; i < buckets.size(); i ++){
+            double totalDistance = calculateDistance(pixel, buckets.get(i));
+            // Evaluates total distance against minimum distance for given bucket
+            if (totalDistance < minDistance) {
+                minDistance = totalDistance;
+                bestColor = i;
+            }
+        }
+        // Add 1 to the bucket closest to pixel color
+        bucketCounter[bestColor] ++;
+    }
+
+    abstract double calculateDistance(int pixel, C bucket);
+
+    /**
+     * Combine list of color-buckets and bucketCount int[] into a map. The list holds the keys and the int[] holds the values.
+     * The arrays get combined by index number.
+     * @param buckets list of implemented type with keys of the map.
+     * @param bucketCount int[] with the values for the map.
+     * @return a map with key-value pairs from the input list and int[].
```

Comment:
Focus on the concepts (palette colors, color count...) rather than the technical representation (list, int[]). The technical stuff is already there in the code - the reader needs to understand what it represents.